server/src/main/java/org/elasticsearch/index/mapper/NodeMappingStats.java

```diff
@@ -63,17 +76,31 @@ public ByteSizeValue getTotalEstimatedOverhead() {
         return ByteSizeValue.ofBytes(totalEstimatedOverhead);
     }
 
+    public long getTotalSegments() {
+        return totalSegments;
+    }
+
+    public long getTotalSegmentFields() {
+        return totalSegmentFields;
+    }
+
     @Override
     public void writeTo(StreamOutput out) throws IOException {
         out.writeVLong(totalCount);
         out.writeVLong(totalEstimatedOverhead);
+        if (out.getTransportVersion().onOrAfter(TransportVersions.SEGMENT_FIELDS_STATS)) {
+            out.writeVLong(totalSegments);
+            out.writeVLong(totalSegmentFields);
+        }
     }
 
     @Override
     public XContentBuilder toXContent(XContentBuilder builder, Params params) throws IOException {
         builder.startObject(Fields.MAPPINGS);
         builder.field(Fields.TOTAL_COUNT, getTotalCount());
         builder.humanReadableField(Fields.TOTAL_ESTIMATED_OVERHEAD_IN_BYTES, Fields.TOTAL_ESTIMATED_OVERHEAD, getTotalEstimatedOverhead());
+        builder.field(Fields.TOTAL_SEGMENT_FIELDS, totalSegments);
```

Comment:
The field name suggests this should be the following perhaps?

```suggestion
        builder.field(Fields.TOTAL_SEGMENT_FIELDS, totalSegmentFields);
```

Reply:
Good catch!